execution/contexts/bond_context.go

```diff
@@ -0,0 +1,178 @@
+package contexts
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/hyperledger/burrow/acm"
+	"github.com/hyperledger/burrow/acm/acmstate"
+	"github.com/hyperledger/burrow/acm/validator"
+	"github.com/hyperledger/burrow/execution/exec"
+	"github.com/hyperledger/burrow/logging"
+	"github.com/hyperledger/burrow/permission"
+	"github.com/hyperledger/burrow/txs/payload"
+)
+
+type BondContext struct {
+	StateWriter  acmstate.ReaderWriter
+	ValidatorSet validator.ReaderWriter
+	Logger       *logging.Logger
+	tx           *payload.BondTx
+}
+
+// Execute a BondTx to add a new validator
+func (ctx *BondContext) Execute(txe *exec.TxExecution, p payload.Payload) error {
+	var ok bool
+	ctx.tx, ok = p.(*payload.BondTx)
+	if !ok {
+		return fmt.Errorf("payload must be BondTx, but is: %v", txe.Envelope.Tx.Payload)
+	}
+
+	// the account initiating the bond
+	account, err := ctx.StateWriter.GetAccount(ctx.tx.Input.Address)
+	if err != nil {
+		return err
+	}
+
+	// ensure pubKey of validator is set
+	val := ctx.tx.Validator
```

Comment:
I don't think this is the correct type. We use `TemplateAccount` for `GovTx` because it can make arbitrary changes to an account. I think this should just be `PublicKey`

Reply:
Updated